api/src/main/java/io/strimzi/api/kafka/model/ExternalLogging.java

```diff
@@ -18,25 +20,41 @@
 )
 @JsonPropertyOrder({"type", "name"})
 @JsonInclude(JsonInclude.Include.NON_NULL)
+// TODO remove this Note OneOf constraint declared on superclass
 public class ExternalLogging extends Logging {
 
     private static final long serialVersionUID = 1L;
     public static final String TYPE_EXTERNAL = "external";
 
     private String name;
 
+    private ExternalConfigurationReference valueFrom;
+
     @Description("Must be `" + TYPE_EXTERNAL + "`")
     @Override
     public String getType() {
         return TYPE_EXTERNAL;
     }
 
     @Description("The name of the `ConfigMap` from which to get the logging configuration.")
+    @DeprecatedProperty(description = "Replaced with valueFrom", removalVersion = "v1beta2")
+    @Deprecated
+    @PresentInVersions("v1alpha1-v1beta1")
```

Comment:
This means that there is no release which supports both in parallel. You expect users to move to `v1beta2` in the same release you introduce the new field. Is that really what we want?

Reply:
`valueFrom` is annotated `v1alpha1+`, so would be available in `v1alpha1`, `v1beta1` and `v1beta2`. OTOH `name` would only be available in available in `v1alpha1`, `v1beta1`. So `v1alpha1`, `v1beta1` are in common.

Reply:
But not on practical level because after the migration to CRD v1 you would practically use only v1beta2, or?

Reply:
Yes, but `v1alpha1`, `v1beta1` don't have structural schemas, so CRD v1 implies only using v1beta2 resources anyway.